testdata/course-languages/bash/grader.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Compile the project and run the grader.
+
+readonly THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly DEFAULT_OUTPUT_PATH="${THIS_DIR}/../output/result.json"
+
+function main() {
+    if [[ $# -ne 0 ]]; then
+        echo "USAGE: $0"
+        return 1
+    fi
+
+    trap exit SIGINT
+
+    cd "${THIS_DIR}"
+
+    # Allow the grader to run locally by changing the output location
+    # if not in the docker image.
+    local outputPath="${DEFAULT_OUTPUT_PATH}"
+    if [[ ! -d $(dirname "${outputPath}") ]] ; then
+        outputPath="$(basename "${outputPath}")"
+    fi
+    
+    # Run grader.
+    grade > "${outputPath}"
```

Comment:
Instead of echoing the output in the function and redirecting it here,
I would pass in `outputPath` to the function and write it out in `grade()`.
In this way, a student can't echo in their own code and mess up the grader.